src/hypothesis/strategies.py

```diff
@@ -1174,11 +1191,10 @@ class ChoiceStrategy(SearchStrategy):
         def do_draw(self, data):
             return Chooser(current_build_context(), data)
 
-    return ReprWrapperStrategy(
```

Comment:
It doesn't, because this PR adds a `@defines_strategy` decorator for the laziness (so that the deprecation warning/error happens at test execution time rather than definition time), which sets the repr appropriately. so the `ReprWrapperStrategy` is no longer needed and doesn't do anything.